packages/account/src/providers/transaction-request/script-transaction-request.ts

```diff
@@ -39,10 +40,7 @@ export interface ScriptTransactionRequestLike extends BaseTransactionRequestLike
  */
 export class ScriptTransactionRequest extends BaseTransactionRequest {
   static from(obj: ScriptTransactionRequestLike) {
-    if (obj instanceof this) {
-      return obj;
-    }
-    return new this(obj);
+    return new this(clone(obj));
```

Comment:
Should we have some simple tests on all these `from` methods?

 1. Clone source object/instance
 1. Modify source object/instance
 1. Validate if the new instance remained pristine

I'd assume we have no tests like these, or we'd have caught the issue earlier.

Reply:
Good point ee922c837